advanced_alchemy/repository/_async.py

```diff
@@ -512,6 +513,7 @@ async def get_or_upsert(
                 empty, all fields are matched.
             upsert: When using match_fields and actual model values differ from
                 `kwargs`, perform an update operation on the model.
+            create: Should a model be created.  If no model is found, an exception is raised.
```

Comment:
OK, sorry I'm starting to get a bit confused.

We call it `get_or_upsert()` where "upsert" means to update existing or insert new. However, "get_or_" implies that if we can get an instance, then we don't upsert. But then that means that if we can get an instance, we can't upsert because we've would have been able to "get" it.

Now adding a flag to disable creation means we can disable the "sert" part of the upsert as well.
Is this a sign that we are jamming too much functionality in this method??

(Sorry if I'm missing context, but no linked issue or description on the PR.)

Reply:
@JacobCoffee you want to explain your use case on this one?

Reply:
@peterschutt , I think we need to change the `upsert` attribute to `perform_update` or something similar.  Then the addition of the `create` or `perform_create` operation would make more sense.

If I understood @JacobCoffee use case, he wanted to fetch and update.  But he didn't want to create an object if it doesn't exist.

Reply:
I also added a bit of context for the change

Reply:
My use case here is I want to get something and update it only. I made a `get_and_update` method in my work project but @cofin  (and I? cant remember) thought it would be better to just add on to the existing `get_and_upsert` with the "add if not exist" turned off. 

It maybe does feel crammed in, but I'm not sure if I mind? I'm good for whichever path we decide

Reply:
> wanted to fetch and update

Sounds like a good method name :)

Imagine you are a future reader of this code, what is clearer:

- `repo.fetch_and_update(...)` something along these lines
- `repo.get_or_upsert(data, create=False)`.

Reply:
> `repo.fetch_and_update(...)`

I like this even if it adds more methods to look up/remember

Reply:
To really argument to any of this.   To be consistent with naming, I recommend we go with `get_and_update`.  We don't use the `fetch` verb anywhere else, so it may be confusing.